webapp/src/components/SortListButton/SortListButton.vue

```diff
@@ -0,0 +1,20 @@
+<template>
+     <button  @click.prevent="sortlist">{{buttontext}}</button>
+</template>
+<script>
+export default {
+    name: "SortListButton",
+    data(){
+        return {
+            buttontext: "ascending"
+        }
+    },
+    methods:{
+        sortlist(){
+            this.$emit("sortorder", this.buttontext)
+            //ternärer Ausdruck
+            this.buttontext = this.buttontext=="ascending"?"descending":"ascending"
+        }
```

Comment:
This is obviously a computed property:

```suggestion
        sortlist(){
            this.$emit("sortorder", this.buttontext)
        }
      },
      computed: {
        buttontext() {
            return this.buttontext === 'ascending' ? 'descending' : 'ascending'
        }
```